### DISCUSSION TITLE
specify user id for the web app

### DISCUSSION THREAD

**Discussion by FBR65** [langgenius/dify#8479] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Would it be possible to specify the user_id or a user id to the Web App  like

http://xxx.xxx.xxx.xxx/chatbot/rMHaJuDPCvi6zHww?user_id=emil or
http://xxx.xxx.xxx.xxx/chatbot/rMHaJuDPCvi6zHww?user=emil

Thanx

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by nanchengHorizon** [comment]
I also have this problem, I would like to ask if it is solved

**Comment by zy0606** [comment]
Has this been resolved? I'm experiencing the same problem.

**Reply by nanchengHorizon** [reply]
Source code cannot be solved! My solution: rewrite the demo directly, then manually write the userid logic, and then iterate on the demo version

**Reply by MrWangChong** [reply]
> Source code cannot be solved! My solution: rewrite the demo directly, then manually write the userid logic, and then iterate on the demo version

Can you send me how you made the changes? I also encountered the same problem

**Comment by nanchengHorizon** [comment]
At page initialization, get the **session_id** from the **url** and store it in the **utils common.ts** user variable. I'm sorry I can only give you ideas, not code. If you are not familiar with the code, you can use **Cursor** or **claude** to do this.

**Comment by tipani86** [comment]
It took a while to figure out but this conversation here includes a verifiable solution. Although some posts are about embedding the page in iframe, the same method also works in "raw" URL when you use the published app directly on a browser: https://github.com/langgenius/dify/issues/26049

tl;dr

It includes encoding the user_id in a specific way described in the issue linked above, then feeding it in as URL parameter `sys.user_id`

See below screenshot for reference. The LLM is instructed to print out the sys.user_id at the very end of its response. In below example, the encoded URL parameter corresponds to the `user_12345` string.

<img width="1921" height="741" alt="image" src="https://github.com/user-attachments/assets/4af70f32-0466-407b-90f3-85f076cd16b2" />
